hyperspy/roi.py

```diff
@@ -486,6 +490,28 @@ def __call__(self, signal, out=None, axes=None):
         return s
 
 
+def guess_vertical_or_horizontal(axes, signal):
+    # Figure out whether to use horizontal or veritcal line:
+    if axes[0].navigate:
+        plotdim = len(signal._plot.navigator_data_function().shape)
+        axdim = signal.axes_manager.navigation_dimension
+        idx = signal.axes_manager.navigation_axes.index(axes[0])
+    else:
+        plotdim = len(signal._plot.signal_data_function().shape)
+        axdim = signal.axes_manager.signal_dimension
+        idx = signal.axes_manager.signal_axes.index(axes[0])
+
+    if plotdim == 2:  # Plot is an image
+        # axdim == 1 and plotdim == 2 indicates "spectrum stack"
+        if idx == 0 and axdim != 1:    # Axis is horizontal
+            return "vertical"
+        else:  # Axis is vertical
+            return "horizontal"
+    elif plotdim == 1:  # It is a spectrum
+        return "vertical"
+    else:
+        raise ValueError("Could not find valid widget type")
```

Comment:
Similar to below: more clear error message mentioning that the axes provided doesn't allow to determine the appropriate widget?

Reply:
Done